gen/__init__.py

```diff
@@ -306,20 +305,67 @@ def do_gen_package(config, package_filename):
     log.info("Package filename: %s", package_filename)
 
 
-def extract_files_with_path(start_files, paths):
+def render_late_content(content, late_values):
```

Comment:
talked about this in slack, but this is doing the same thing as https://github.com/dcos/dcos/blob/master/gen/build_deploy/azure.py#L23 and https://github.com/dcos/dcos/blob/master/gen/build_deploy/aws.py#L230, would be cool to make them all use one function which is tested and more readable in the future.